js-src/compat/plugins/customer-email-verification/checkout-customer-email-verification.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Checkout scripts for: Customer Email Verification (by zorem).
+ */
+
+(function (root, factory) {
+	if ( typeof define === 'function' && define.amd ) {
+		define([], factory(root));
+	} else if ( typeof exports === 'object' ) {
+		module.exports = factory(root);
+	} else {
+		root.CheckoutCustomerEmailVerification = factory(root);
+	}
+})(typeof global !== 'undefined' ? global : this.window || this.global, function (root) {
+
+	'use strict';
+
+	var $ = jQuery;
+	var _hasJQuery = ( $ != null );
+
+	var _hasInitialized = false;
+	var _publicMethods = {};
+	var _settings = {
+		emailFieldSelector: '#billing_email',
+	}
+
+
+
+	/**
+	 * METHODS
+	 */
+
+
+
+	/**
+	 * Validate email field.
+	 * 
+	 * @param Field  field  The email field to validate.
+	 * @param Event  e      The event object.
+	 */
+	var validateEmailField = function( field, e ) {
+		// Trigger checkout update before validation
+		$( document.body ).trigger( 'update_checkout' );
+
+		// Validate email field after a delay
+		setTimeout( function() {
+			CheckoutValidation.validateField( field, e.type );
+		}, 1000 );
+	}
```

Comment:
@Vlad, is the fields supposed to be re-validated after the `update_checkout` process completes? If so, the delay of 1 second might not be enough for some websites, and might be too much for faster websites.

Perhaps using the event `updated_checkout` would be a better approach, as this would run after the checkout fragments are replaced. Use a code similar to this below:

``` 
$( document.body ).on( 'updated_checkout', function(){ /* your code */ } );
```